### ISSUE TITLE
Kubelet leaks /run/systemd/inhibit/X.ref when exec iptables triggering selinux

### ISSUE THREAD

**Issue by champtar** [kubernetes/kubernetes#106292]
### What happened?

```
type=AVC msg=audit(1636522420.778:201): avc:  denied  { write } for  pid=8920 comm="iptables" path="/run/systemd/inhibit/2.ref" dev="tmpfs" ino=58429 scontext=system_u:system_r:iptables_t:s0 tcontext=system_u:object_r:systemd_logind_inhibit_var_run_t:s0 tclass=fifo_file permissive=1


type=SYSCALL msg=audit(1636522420.778:201): arch=x86_64 syscall=execve success=yes exit=0 a0=c0012aa2e8 a1=c0005aab90 a2=c000fd3d00 a3=8 items=0 ppid=7642 pid=8920 auid=4294967295 uid=0 gid=0 euid=0 suid=0 fsuid=0 egid=0 sgid=0 fsgid=0 tty=(none) ses=4294967295 comm=iptables exe=/usr/sbin/xtables-nft-multi subj=system_u:system_r:iptables_t:s0 key=(null)
```
ppid 7642 == kubelet

```
# ls -lia /proc/7642/fd
...
16134 l-wx------. 1 root root 64 Nov 10 05:32 26 -> /run/systemd/inhibit/2.ref

# cat /proc/7642/fdinfo/26 
pos:	0
flags:	0104001
mnt_id:	28
```
https://unix.stackexchange.com/questions/639998/how-to-interpret-octal-flags-in-proc-pid-fdinfo-n

### What did you expect to happen?

iptables should not have the /run/systemd/inhibit/X.ref fd passed to him
we can:
- set close on exec flag
- close all fds when we exec programs
- both

### How can we reproduce it (as minimally and precisely as possible)?

Centos 8.4 with permissive selinux
kubelet 1.22.3 with shutdownGracePeriod / shutdownGracePeriodCriticalPods set

### Kubernetes version

```console
# kubectl version
Client Version: version.Info{Major:"1", Minor:"22", GitVersion:"v1.22.3", GitCommit:"c92036820499fedefec0f847e2054d824aea6cd1", GitTreeState:"clean", BuildDate:"2021-10-27T18:41:28Z", GoVersion:"go1.16.9", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"22", GitVersion:"v1.22.3", GitCommit:"c92036820499fedefec0f847e2054d824aea6cd1", GitTreeState:"clean", BuildDate:"2021-10-27T18:35:25Z", GoVersion:"go1.16.9", Compiler:"gc", Platform:"linux/amd64"}
```

### Cloud provider

none

### OS version

```console
Centos 8.4
```

### Install tools

kubeadm

### Container runtime (CRI) and and version (if applicable)

containerd 1.5.7

**Comment by champtar** [comment]
Fixed in 1.23+ as go was updated

**Comment by rptaylor** [comment]
@champtar do you have any other specific details about the fix, which minor version, a commit or PR, etc ?

**Comment by champtar** [comment]
This is fixed in k8s 1.23+ because it uses golang 1.17+, which include https://github.com/golang/go/commit/e97d8eb027c0067f757860b6f766644de15941f2
There will not be any fixes in 1.22: https://github.com/kubernetes/kubernetes/pull/106293#issuecomment-978888322
